plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -508,6 +512,18 @@ private static Optional<Object> translateValue(Object trinoNativeValue, Type typ
             return Optional.of(trinoNativeValue);
         }
 
+        if (type == DATE) {
+            return Optional.of(new Date(TimeUnit.DAYS.toMillis((Long) trinoNativeValue)));
```

Comment:
I would recommend adding a preparatory commit to use `LocalDate` for DATE type in other places.